File: app/src/main/java/com/commit451/dialup/sample/MainActivity.java
package com.commit451.dialup.sample;

import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.ViewGroup;
import android.widget.TextView;

import com.commit451.dialup.DialUp;

import io.reactivex.Observer;
import io.reactivex.disposables.Disposable;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final ViewGroup root = (ViewGroup) findViewById(R.id.root);

        final TextView textView = (TextView) findViewById(R.id.status);
        DialUp.listen(this)
                .subscribe(new Observer<Boolean>() {
                    @Override
                    public void onSubscribe(Disposable d) {
                    }

                    @Override
                    public void onNext(Boolean aBoolean) {
                        Snackbar.make(root, "Status changed to " + aBoolean, Snackbar.LENGTH_LONG)
                                .show();
                        textView.setText("Connected: " + aBoolean);
                    }

                    @Override
                    public void onError(Throwable e) {
                        Log.e("TAG", "fail", e);
                    }

                    @Override
                    public void onComplete() {
                    }
                });

    }
}

File: app/src/main/java/com/commit451/dialup/sample/MainActivity.kt
package com.commit451.dialup.sample

import android.os.Bundle
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.commit451.dialup.DialUp
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val root = findViewById<ViewGroup>(R.id.root)

        val textView = findViewById<TextView>(R.id.status)
        DialUp.observable(this)
                .subscribe { connected ->
                    Snackbar.make(root, "Connectivity changed to: $connected", Snackbar.LENGTH_LONG)
                            .show()
                    textView.text = "Connected: $connected"
                }
    }
}
